Add SetExplicitRadixPoint config to json.Visitor

carsonip opened the pull request:
SetExplicitRadixPoint configures whether the visitor encodes floating point values with an explicit radix point.
By default, equiv to SetExplicitRadixPoint(false), the radix point will be skipped if it is not needed.
e.g. 1.0 to 1 instead of 1.0, 100000000 to 1e+8 instead of 1.0e+8.
If true is passed, the encoded number will always contain a radix point,
in either decimal form or scientific notation.
This may be useful to signal the type of the number to a json parser.

Related issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34680

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/14)
* Commit: 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/14","number":14,"commit":"871d308355414f2c24399a5c6b68d238156ae29b"}],"number":14}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/16)
* Commit: c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6

### History
* :green_heart: [Build #14](https://buildkite.com/elastic/go-structform/builds/14) succeeded 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/16","number":16,"commit":"c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/14","number":14,"commit":"871d308355414f2c24399a5c6b68d238156ae29b"}],"number":16}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/17)
* Commit: 5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f

### History
* :green_heart: [Build #16](https://buildkite.com/elastic/go-structform/builds/16) succeeded c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6
* :green_heart: [Build #14](https://buildkite.com/elastic/go-structform/builds/14) succeeded 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/17","number":17,"commit":"5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/16","number":16,"commit":"c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/14","number":14,"commit":"871d308355414f2c24399a5c6b68d238156ae29b"}],"number":17}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/18)
* Commit: ad45eb9d6fe3a4702c908829a4de3cdc62de52dc

### History
* :green_heart: [Build #17](https://buildkite.com/elastic/go-structform/builds/17) succeeded 5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f
* :green_heart: [Build #16](https://buildkite.com/elastic/go-structform/builds/16) succeeded c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6
* :green_heart: [Build #14](https://buildkite.com/elastic/go-structform/builds/14) succeeded 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/18","number":18,"commit":"ad45eb9d6fe3a4702c908829a4de3cdc62de52dc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/17","number":17,"commit":"5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/16","number":16,"commit":"c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/14","number":14,"commit":"871d308355414f2c24399a5c6b68d238156ae29b"}],"number":18}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/21)
* Commit: def82ecec1977447a3c8af3a78c233deaeaa01b5

### History
* :green_heart: [Build #20](https://buildkite.com/elastic/go-structform/builds/20) succeeded aa3690a7c4b57b2789d38cc573542c90806d4b42
* :green_heart: [Build #18](https://buildkite.com/elastic/go-structform/builds/18) succeeded ad45eb9d6fe3a4702c908829a4de3cdc62de52dc
* :green_heart: [Build #17](https://buildkite.com/elastic/go-structform/builds/17) succeeded 5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f
* :green_heart: [Build #16](https://buildkite.com/elastic/go-structform/builds/16) succeeded c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6
* :green_heart: [Build #14](https://buildkite.com/elastic/go-structform/builds/14) succeeded 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/21","number":21,"commit":"def82ecec1977447a3c8af3a78c233deaeaa01b5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/20","number":20,"commit":"aa3690a7c4b57b2789d38cc573542c90806d4b42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/18","number":18,"commit":"ad45eb9d6fe3a4702c908829a4de3cdc62de52dc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/17","number":17,"commit":"5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/16","number":16,"commit":"c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/14","number":14,"commit":"871d308355414f2c24399a5c6b68d238156ae29b"}],"number":21}
buildkite-pr-comment-->

carsonip committed:
Add SetUnambiguousFloat config to json.Visitor

SetUnambiguousFloat configures how the visitor encodes floating point
values to a json float unambiguously.
By default, equiv to SetUnambiguousFloat(false), the visitor will encode
with the smallest number of digits.
e.g. 1.0 to 1 instead of 1.0, 100000000 to 1e+8 instead of 1.0e+8.
If true is passed, floating point values will always contain a decimal
point, in either decimal form or scientific notation.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/22)
* Commit: ffd0e24f464d1c44eee306b5fec8b722e4c52f32

### History
* :green_heart: [Build #21](https://buildkite.com/elastic/go-structform/builds/21) succeeded def82ecec1977447a3c8af3a78c233deaeaa01b5
* :green_heart: [Build #20](https://buildkite.com/elastic/go-structform/builds/20) succeeded aa3690a7c4b57b2789d38cc573542c90806d4b42
* :green_heart: [Build #18](https://buildkite.com/elastic/go-structform/builds/18) succeeded ad45eb9d6fe3a4702c908829a4de3cdc62de52dc
* :green_heart: [Build #17](https://buildkite.com/elastic/go-structform/builds/17) succeeded 5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f
* :green_heart: [Build #16](https://buildkite.com/elastic/go-structform/builds/16) succeeded c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6
* :green_heart: [Build #14](https://buildkite.com/elastic/go-structform/builds/14) succeeded 871d308355414f2c24399a5c6b68d238156ae29b

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/22","number":22,"commit":"ffd0e24f464d1c44eee306b5fec8b722e4c52f32"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/21","number":21,"commit":"def82ecec1977447a3c8af3a78c233deaeaa01b5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/20","number":20,"commit":"aa3690a7c4b57b2789d38cc573542c90806d4b42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/18","number":18,"commit":"ad45eb9d6fe3a4702c908829a4de3cdc62de52dc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/17","number":17,"commit":"5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/16","number":16,"commit":"c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6"}],"number":22}
buildkite-pr-comment-->

carsonip commented on json/visitor.go:
```diff
@@ -100,6 +101,15 @@ func (v *Visitor) SetIgnoreInvalidFloat(b bool) {
+// e.g. 1.0 to 1 instead of 1.0, 100000000 to 1e+8 instead of 1.0e+8.
+// If true is passed, floating point values will always contain a decimal point,
+// in either decimal form or scientific notation.
+func (v *Visitor) SetUnambiguousFloat(b bool) {
```
[To reviewer] feel free to suggest a better name, e.g. SetAlwaysEmitDecimalPoint, SetAlwaysWriteDecimalPoint

leehinman replied:
maybe `ExplicitRadixPoint`?  https://en.wikipedia.org/wiki/Floating-point_arithmetic

carsonip replied:
thanks for the suggestion. Updated.

leehinman reviewed (commented):
I'm ok with the code.

suggestion on feature name

carsonip committed:
Change to SetExplicitRadixPoint

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/go-structform/builds/23)
* Commit: dfdc38b07991e10bf937f3d4c891fb85377add83

### History
* :green_heart: [Build #22](https://buildkite.com/elastic/go-structform/builds/22) succeeded ffd0e24f464d1c44eee306b5fec8b722e4c52f32
* :green_heart: [Build #21](https://buildkite.com/elastic/go-structform/builds/21) succeeded def82ecec1977447a3c8af3a78c233deaeaa01b5
* :green_heart: [Build #20](https://buildkite.com/elastic/go-structform/builds/20) succeeded aa3690a7c4b57b2789d38cc573542c90806d4b42
* :green_heart: [Build #18](https://buildkite.com/elastic/go-structform/builds/18) succeeded ad45eb9d6fe3a4702c908829a4de3cdc62de52dc
* :green_heart: [Build #17](https://buildkite.com/elastic/go-structform/builds/17) succeeded 5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f
* :green_heart: [Build #16](https://buildkite.com/elastic/go-structform/builds/16) succeeded c23d2ce2e81f1862f41aeacb99a1657f8e2e8ea6

<!--buildkite-pr-comment-go-structform
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/23","number":23,"commit":"dfdc38b07991e10bf937f3d4c891fb85377add83"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/22","number":22,"commit":"ffd0e24f464d1c44eee306b5fec8b722e4c52f32"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/21","number":21,"commit":"def82ecec1977447a3c8af3a78c233deaeaa01b5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/20","number":20,"commit":"aa3690a7c4b57b2789d38cc573542c90806d4b42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/18","number":18,"commit":"ad45eb9d6fe3a4702c908829a4de3cdc62de52dc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/go-structform/builds/17","number":17,"commit":"5fd9f62d90e92e3d5bd6c2437c8ea95af2feba3f"}],"number":23}
buildkite-pr-comment-->

leehinman reviewed (approved):
LGTM

carsonip commented:
@mauri870 @leehinman I don't have the merge button. Please merge and release as you see fit.
